WorkspaceServer.Tests/Kernel/CSharpKernelTests.cs

```diff
@@ -464,5 +465,75 @@ public async Task OnLoadAsync(IKernel kernel)
                         .ContainSingle(e => e.Value is CodeSubmissionEvaluated &&
                                             e.Value.As<CodeSubmissionEvaluated>().Code.Contains("using System.Reflection;"));
         }
+
+        [Fact]
+        public async Task Loads_native_dependencies_from_nugets()
+        {
+            var kernel = new CompositeKernel
+            {
+                new CSharpKernel().UseNugetDirective(new NativeAssemblyLoadHelper())
+            };
+
+            var command = new SubmitCode(@"#r ""nuget:Microsoft.ML, 1.3.1""
+using Microsoft.ML;
+using Microsoft.ML.Data;
+using System;
+
+class IrisData
+        {
+            public IrisData(float sepalLength, float sepalWidth, float petalLength, float petalWidth)
+            {
+                SepalLength = sepalLength;
+                SepalWidth = sepalWidth;
+                PetalLength = petalLength;
+                PetalWidth = petalWidth;
+            }
+            public float SepalLength;
+            public float SepalWidth;
+            public float PetalLength;
+            public float PetalWidth;
+        }
+
+        var data = new[]
+        {
+    new IrisData(1.4f, 1.3f, 2.5f, 4.5f),
+    new IrisData(2.4f, 0.3f, 9.5f, 3.4f),
+    new IrisData(3.4f, 4.3f, 1.6f, 7.5f),
+    new IrisData(3.9f, 5.3f, 1.5f, 6.5f),
+};
+
+        MLContext mlContext = new MLContext();
+        var pipeline = mlContext.Transforms
+            .Concatenate(""Features"", ""SepalLength"", ""SepalWidth"", ""PetalLength"", ""PetalWidth"")
+            .Append(mlContext.Clustering.Trainers.KMeans(""Features"", numberOfClusters: 2));
+
+try
+{
+    pipeline.Fit(mlContext.Data.LoadFromEnumerable(data));
+    Console.WriteLine(""success"");
+}
+catch (Exception e)
+{
+    Console.WriteLine(e);
+}", "csharp");
+
+            var result = await kernel.SendAsync(command);
+
+            var events = result.KernelEvents
```

Comment:
This could hang forever.

I added functionality to make this kind of test easier. Here's an example:

https://github.com/dotnet/try/blob/7d1dd593bab0172a981b7f8618960dc105ffdae1/Microsoft.DotNet.Interactive.Jupyter.Tests/MagicCommandTests.cs#L42

Reply:
I don't think this works for commands that do async work? Kernel.SendAsync returning doesn't mean the command has done anything yet. 

Note that I did try this and my test fails because the assertions run before anything has happened.

Reply:
If that's the case it's safest to provide a timeout.

Reply:
Actually, this should work in any case because the events are available on the `KernelEvents` observable even after they were dispatched, precisely to make this sort of thing easier.